plugins/woocommerce/tests/api-core-tests/ci-release.global-setup.js

```diff
@@ -0,0 +1,109 @@
+const { UPDATE_WC, USER_KEY, USER_SECRET } = process.env;
+const { test, expect } = require( '@playwright/test' );
+const path = require( 'path' );
+const fs = require( 'fs' );
+
+const zipPath = path.resolve( 'tmp', 'woocommerce.zip' );
+const downloadURL = `https://github.com/woocommerce/woocommerce/releases/download/${ UPDATE_WC }/woocommerce.zip`;
+
+test( `Setup remote test site`, async ( { page, request } ) => {
+	await test.step( `Download WooCommerce build zip`, async () => {
+		const response = await request.get( downloadURL );
+		expect( response.ok() ).toBeTruthy();
+		const body = await response.body();
+		fs.mkdirSync( 'tmp', { recursive: true } );
+		fs.writeFileSync( zipPath, body );
+	} );
+
+	await test.step( 'Login to wp-admin', async () => {
+		const Username = 'Username or Email Address';
+		const Password = 'Password';
+		const Log_In = 'Log In';
+		const Dashboard = 'Dashboard';
+
+		// Need to wait until network idle. Otherwise, Password field gets auto-cleared after typing password in.
+		await page.goto( '/wp-admin', { waitUntil: 'networkidle' } );
+		await page.getByLabel( Username ).fill( USER_KEY );
+		await page.getByLabel( Password, { exact: true } ).fill( USER_SECRET );
+		await page.getByRole( 'button', { name: Log_In } ).click();
+		await expect(
+			page
+				.locator( '#menu-dashboard' )
+				.getByRole( 'link', { name: Dashboard } )
+		).toBeVisible();
+	} );
+
+	await test.step( `Deactivate currently installed WooCommerce version`, async () => {
+		const response = await request.put(
+			'/wp-json/wp/v2/plugins/woocommerce/woocommerce',
+			{
+				data: {
+					status: 'inactive',
+				},
+			}
+		);
+		expect( response.ok() ).toBeTruthy();
+	} );
+
+	await test.step( `Delete currently installed WooCommerce version`, async () => {
+		const response = await request.delete(
+			'/wp-json/wp/v2/plugins/woocommerce/woocommerce'
+		);
+		expect( response.ok() ).toBeTruthy();
+	} );
+
+	await test.step( `Install WooCommerce ${ UPDATE_WC }`, async () => {
+		const Upload_Plugin = 'Upload Plugin';
+		const Plugin_zip_file = 'Plugin zip file';
+		const Install_Now = 'Install Now';
+		const Activate_Plugin = 'Activate Plugin';
+
+		await page.goto( '/wp-admin/plugin-install.php' );
+		await page.getByRole( 'button', { name: Upload_Plugin } ).click();
+		await page.getByLabel( Plugin_zip_file ).setInputFiles( zipPath );
+		const responsePromise = page.waitForResponse(
+			'**/wp-admin/update.php?action=upload-plugin'
+		);
+		await page.getByRole( 'button', { name: Install_Now } ).click();
+		const uploadResponse = await responsePromise;
```

Comment:
> Wouldn't it work here just to handle the page.waitForResponse with await instead of doing that in line 68?

I think it would still work. The reason I separated line 64 and line 68 was because I was just following the format [here](https://playwright.dev/docs/api/class-page#page-wait-for-response). 

I had just pushed a commit that combines those two lines into a single `await page.waitForResponse()` call -- 6e66f48af89b48ab5dad018370821d6c8c7d6a05